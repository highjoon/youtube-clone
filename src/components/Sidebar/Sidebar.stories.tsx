import Sidebar from './Sidebar';

export default {
  component: Sidebar,
  title: 'components/Sidebar',
};

export const HomeSidebar = () => <Sidebar icon={'Home'} />;

export const TrendsDefaultSidebar = () => <Sidebar icon={'TrendsDefault'} />;

export const SubscriptionsSidebar = () => <Sidebar icon={'Subscriptions'} />;
