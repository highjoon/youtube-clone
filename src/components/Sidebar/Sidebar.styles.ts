import styled from '@emotion/styled';

export const Container = styled.div`
  width: 240px;
  height: 40px;
  display: flex;
  align-items: center;
  padding: 11px 24px;
  gap: 23px;
  cursor: pointer;

  &:hover {
    background: var(--color-black-dark4);

    span {
      font-weight: 600;
    }
  }
`;

export const Title = styled.span`
  font-size: 14px;
  line-height: 16px;
  color: var(--color-primary-black);
`;
