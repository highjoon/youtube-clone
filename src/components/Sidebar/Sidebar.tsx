import React, { useState } from 'react';

import Icon from '@/components/Icon/Icon';

import * as Styled from './Sidebar.styles';

interface Props {
  className?: string;
  icon: 'Home' | 'TrendsDefault' | 'Subscriptions';
}

const Sidebar = ({ className, icon }: Props) => {
  const [isHovered, setIshovered] = useState<boolean>(false);

  const iconColor = isHovered ? 'var(--color-primary-red)' : 'var(--color-black-dark2)';

  return (
    <Styled.SidebarWrapper
      className={className}
      onMouseEnter={() => setIshovered(true)}
      onMouseLeave={() => setIshovered(false)}
    >
      <Icon icon={icon} color={iconColor} />
      <Styled.Title>{icon}</Styled.Title>
    </Styled.SidebarWrapper>
  );
};

export default Sidebar;
