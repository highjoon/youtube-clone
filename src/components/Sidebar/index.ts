import { Container, MenuIcon, Title } from './Sidebar';

const Sidebar = Object.assign(Container, {
  Title,
  Icon: MenuIcon,
});

export default Sidebar;
