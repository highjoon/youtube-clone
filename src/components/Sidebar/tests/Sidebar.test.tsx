import { matchers } from '@emotion/jest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Sidebar from '../';

expect.extend(matchers);

describe('Sidebar', () => {
  const SidebarComponent = (
    <Sidebar>
      <Sidebar.Icon icon={'Home'} />
      <Sidebar.Title>Home</Sidebar.Title>
    </Sidebar>
  );

  it('정상적으로 렌더링됩니다.', () => {
    const { asFragment } = render(SidebarComponent);

    expect(asFragment()).toMatchSnapshot();
  });

  it('마우스 hover 시 Container 색상이 var(--color-black-dark4)로 변합니다.', () => {
    const user = userEvent.setup();

    render(SidebarComponent);

    const component = screen.getByTestId('container');

    user.hover(component);
    expect(component).toHaveStyleRule('background', 'var(--color-black-dark4)', {
      target: ':hover',
    });
  });
});
