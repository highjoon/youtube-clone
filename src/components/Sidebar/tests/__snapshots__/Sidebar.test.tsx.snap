// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar 정상적으로 렌더링됩니다. 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 240px;
  height: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 11px 24px;
  gap: 23px;
  cursor: pointer;
}

.emotion-0:hover {
  background: var(--color-black-dark4);
}

.emotion-0:hover span {
  font-weight: 600;
}

.emotion-1 {
  font-size: 14px;
  line-height: 16px;
  color: var(--color-primary-black);
}

<div
    class="emotion-0"
    data-testid="container"
  >
    <icon-mock
      classname="css-js4mbw-Icon"
      data-testid="icon"
    />
    <span
      class="emotion-1"
      data-testid="title"
    >
      Home
    </span>
  </div>
</DocumentFragment>
`;
